core/routing/routing.go

```diff
@@ -36,6 +37,16 @@ type ContentRouting interface {
 	FindProvidersAsync(context.Context, cid.Cid, int) <-chan peer.AddrInfo
 }
 
+// ContentRoutingMany complements ContentRouting intetface allowing providing several keys at
+// the same time, improving performance.
+type ContentRoutingMany interface {
+	// ProvideMany adds the given keys to the content routing system.
+	ProvideMany(ctx context.Context, keys []multihash.Multihash) error
+
+	// Ready checks if the router is ready to start providing keys.
+	Ready() bool
```

Comment:
This seems unrelated to bulk providing things, it's just that both functions were added to a particular struct (the accelerated DHT client)